templates/EC2_Quickstart.yml

```diff
@@ -0,0 +1,156 @@
+Parameters:
+  NewRelicInfrastructureAgent:
+    Type: String
+    Description: "New Relic's infrastructure monitoring agent is a lightweight executable file that collects data about your hosts. It also forwards data from our on-host integrations to New Relic, as well as log data for log analytics."
+    Default: "New-Relic-infrastructure-monitoring-agent"
+  InstallationType:
+    Description: 'Specify the type of installation. Uninstall and resinstall -  The application is taken offline until the reinstallation process completes. In-place update -  The application is available while new or updated files are added to the installation.'
+    Type: String
+    Default: 'Uninstall and reinstall'
+    AllowedValues:
+      - 'Uninstall and reinstall'
+      - 'In-place update'
+  Action:
+    Description: 'Specify whether or not to install or uninstall the package.'
+    Type: String
+    Default: 'Install'
+    AllowedValues:
+      - 'Install'
+      - 'Uninstall'
+  NewRelicLicenseKey:
+    Type: String
+    NoEcho: true
+    Description: The license key associated with the account you wish to export metrics to
+    AllowedPattern: '^[A-Za-z0-9]{40}$'
+  AdditionalParametersLicenseKey:
+    NoEcho: true
+    AllowedPattern: '\{"SSM_NR_LICENSE_KEY": "[A-Za-z0-9]{40}"\}'
+    Description: 'NewRelicLicenseKey in this format - {"SSM_NR_LICENSE_KEY": "<your_license_key>"}'
+    Type: String
+  NewRelicLogsEndpoint:
+    Type: String
+    Description: 'New Relic Logs endpoint URL. Use default unless your New Relic account uses an EU data center. See NR_LOGGING_ENDPOINT in https://docs.newrelic.com/docs/logs/forward-logs/aws-lambda-sending-cloudwatch-logs/ for details.'
+    Default: 'https://log-api.newrelic.com/log/v1'
+    AllowedValues:
+      - 'https://log-api.newrelic.com/log/v1'
+      - 'https://log-api.eu.newrelic.com/log/v1'
+Resources:
+
+  CTNewRelicInstallInfraAgentInstaller:
+    Description: Installs latest version NewRelic Infrastructure agent in EC2 instances once every month.
+    Type: 'AWS::SSM::Association'
+    Properties:
+      AssociationName: CT-NewRelic-Infra-Agent-Installer
+      Name: AWS-ConfigureAWSPackage
+      ScheduleExpression: "rate(30 days)"
+      Targets:
+        - Key: "tag:Supervisor"
+          Values:
+            - "new-relic"
+      Parameters:
+        action: 
+          - !Ref Action
+        installationType: 
+          - !Ref InstallationType
+        name: 
+          - !Ref NewRelicInfrastructureAgent
+        additionalArguments: 
+          - !Ref AdditionalParametersLicenseKey
+
+  NewRelicApiDestination:
+    DependsOn: NewRelicConnection
+    Type: AWS::Events::ApiDestination
+    Properties:
+      Name: CT-NewRelic-Logs-Endpoint
+      HttpMethod: POST

```

Comment:
Is it possible to append any X-Header identifier to this request , so that you can get visibility from NEWR ingestion perspective and measure the success of this integration ?

Reply:
@anshrma 

We've tried adding header parameters in AWS Connection at api destination level but they are not coming to new relic. Could you please help us on this.

Reply:
@anshrma 
Done. We used input transformer feature of event rule to append header tag.
"PackageName": "New-Relic-infrastructure-monitoring-agent"